using-mobile-extensions/adobe-journey-optimizer-decisioning/api-reference.md

```diff
@@ -0,0 +1,542 @@
+                                                                   placementId: @"xcore:offer-placement:1111111111111111" 
+                                                                     itemCount: 2];
+AEPDecisionScope* decisionScope2 = [[AEPDecisionScope alloc] initWithName: @"myScope"];
+
+[AEPMobileOptimize getPropositions: @[decisionScope1, decisionScope2] 
+                        completion: ^(NSDictionary<AEPDecisionScope*, AEPProposition*>* propositionsDict, NSError* error) {
+  if (error != nil) {
+    // handle error   
+    return;
+  }
+
+  AEPProposition* proposition1 = propositionsDict[decisionScope1];
+  // read proposition1 offers
+
+  AEPProposition* proposition2 = propositionsDict[decisionScope2];
+  // read proposition2 offers
+}];
+```
+
+{% endtab %}
+{% endtabs %}
+
+## onPropositionsUpdate
+
+This API registers a permanent callback which is invoked whenever the Edge extension dispatches a response Event received from the Experience Edge Network upon a personalization query. The personalization query requests can be triggered by the `updatePropositions` API, Edge extension `sendEvent` API or launch consequence rules.
+
+{% tabs %}
+{% tab title="iOS — Swift" %}
+### Swift
+
+#### Syntax
+
+```swift
+static func onPropositionsUpdate(perform action: @escaping ([DecisionScope: Proposition]?) -> Void)
+```
+
+* _action_ is invoked with propositions dictionary of type `[DecisionScope: Proposition]`
+
+#### Example
+
+```swift
+Optimize.onPropositionsUpdate { propositionsDict in
+  if let propositionsDict = propositionsDict {
+    // handle propositions
+  }
+}
+```
+
+### Objective-C
+
+#### Syntax
+
+```objc
++ (void) onPropositionsUpdate: (void (^)(NSDictionary<AEPDecisionScope*, AEPProposition*>* _Nullable)) action;
+```
+
+#### Example
+
+```objc
+[AEPMobileOptimize onPropositionsUpdate: ^(NSDictionary<AEPDecisionScope*, AEPProposition*>* propositionsDict) {
+  // handle propositions
+}];
+```
+
+{% endtab %}
+{% endtabs %}
+
+## registerExtension
+
+This `MobileCore` API can be invoked to register the Optimize extension.
+
+{% tabs %}
+{% tab title="iOS — Swift" %}
+### Swift
+
+#### Syntax
+
+```swift
+static func registerExtensions(_ extensions: [NSObject.Type], 
+                               _ completion: (() -> Void)? = nil)
+```
+
+#### Example
+
+```swift
+MobileCore.registerExtensions([Optimize.self, ...]) {
+    // Processing upon registration completion
+}
+```
+
+### Objective-C
+
+#### Syntax
+
+```objc
++ (void) registerExtensions: (NSArray<Class*>* _Nonnull) extensions 
+                 completion: (void (^)(void)) completion;
+```
+
+#### Example
+
+```objc
+[AEPMobileCore registerExtensions:@[AEPMobileOptimize.class, ...] completion:^{
+  // Processing upon registration completion
+}];
+```
+
+{% endtab %}
+{% endtabs %}
+
+## resetIdentities
+
+This `MobileCore` API can also be invoked to clear out the client-side data for Optimize extension, e.g. in-memory propositions cache.
```

Comment:
"... client-side data for the Optimize extension, such as the in-memory propositions cache."